### ISSUE TITLE
unable to import (i'm sure it's something small i'm missing?)

### ISSUE THREAD

**Issue by tukadafoonday** [vincenzosuraci/hassio_meross_sensor_switch#9]
2019-05-07 20:04:22 WARNING (MainThread) [homeassistant.loader] You are using a                   custom integration for meross which has not been tested by Home Assistant. This                   component might cause stability problems, be sure to disable it if you do experi                  ence issues with Home Assistant.
2019-05-07 20:04:33 ERROR (MainThread) [homeassistant.components.hassio] Package                   meross setup failed. Component meross unable to import
2019-05-07 20:12:13 ERROR (MainThread) [homeassistant.components.hassio] Package meross setup failed. Component meross unable to import
2019-05-07 20:17:06 ERROR (MainThread) [homeassistant.components.hassio] Package meross setup failed. Component meross unable to import

**Comment by vincenzosuraci** [comment]
Have you checked that all the files are as described [here](https://github.com/vincenzosuraci/hassio_meross_sensor_switch#install)?

VS

**Comment by tukadafoonday** [comment]
> Have you checked that all the files are as described [here](https://github.com/vincenzosuraci/hassio_meross_sensor_switch#install)?
> 
> VS

Yes, all files are definitely there.

**Comment by tukadafoonday** [comment]
core-ssh:/# hassio homeassistant check --config secrets
Error: Testing configuration at /config
Failed config
  homeassistant.packages.meross.meross:
    - Package meross setup failed. Component meross unable to import
    - meross: [source /config/integrations/meross.yaml:1]
        meross_devices_scan_interval: 900
        password: <!secret>
        scan_interval: 10
        username: <!secret>
Successful config (partial)
  homeassistant.packages.meross.meross:

**Comment by vincenzosuraci** [comment]
Are you in Hassio (RPi)? Which version of HA are you running?

VS

**Comment by tukadafoonday** [comment]
> Are you in Hassio (RPi)? Which version of HA are you running?
> 
> VS

Using Hass.io V0.92.2 on RPi3B+.  I have also split my config.

**Comment by vincenzosuraci** [comment]
I'm not an expert of HAssio, but from here:
```
core-ssh:/# hassio homeassistant check --config secrets
Error: Testing configuration at /config
Failed config
homeassistant.packages.meross.meross:
**- Package meross setup failed. Component meross unable to import
- meross: [source /config/integrations/meross.yaml:1]**
```
it seems that your configuration is failing importing this file: `/config/integrations/meross.yaml`. I do not have such a file in my RPi `config` directory. I do not know if it was you to create it.

Try first to maintain the same yaml configuration files structure (this is just to isolate the source of your problem).

However, another possibility is that, for any reason, Hassio failed to download automatically the required library (meross_iot). In such a case, download it manually from [here](https://github.com/albertogeniola/MerossIot/tree/0.3.X.X). Unzip it and copy the `meross_iot` folder under the following directory in your RPi: `\config\deps\lib\python3.7\site-packages`. Restart Hassio.

VS

**Comment by vincenzosuraci** [comment]
@tukadafoonday did you manage to solve It?

**Comment by tukadafoonday** [comment]
Hey,

Managed to get it loaded by applying it as you mentioned in the install
instructions instead of splitting the config.

It did however put me into a loop is syncing with the meross server and ZHA
timeouts to the extent I could not get HA to boot.

I put it aside for the minute and will get back to it when I have a couple
of the other things setup.

Starting to think that a raspberry Pi might now be powerful enough and I
should upgrade to a NUC or run in a VM on my PC.

Once I load the component again I will post the results and the logs if it
doesn’t work.

Thanks for chasing up.

Andrew

**Comment by tukadafoonday** [comment]
so, in trying to load meross again, it would seem it all tries successully; however, just gets stuck in a I/O loop of timeouts. (see logs)

https://pastebin.com/Z5fmVpfc

am now giving i a go with scan interval at 180s to see how that goes with allowing some other traffic and hopefully at least getting it setup, then can fine tune from there.

********Addition******

Interestingly when I set the scan_interval for 180s I do infact get HA to boot into lovelace and display the Meross switches; however, the logs are loaded with timeouts.

https://pastebin.com/nyStLGBt

Since then I have restarted with a scan_interval of 30seconds which appears to be having a hard to reaching lovelace again.  

It's fair to say I think this is a hardware issue with the I/O of a rpi3b+ not quite good enough when also loading devices with zha.

Thoughts?

**Comment by vincenzosuraci** [comment]
I think your hypothesis Is right. The setup of scanning interval depends on the hardware capabilities. On My Rpi3b+, I setup 10s. With a fiber optic internet connection, I did not experience lags or leaks. 

VS

**Comment by vincenzosuraci** [comment]
What do you mean with "zha"?
I know "ha", "Haas", "Hassio". I miss "zha"...

**Comment by tukadafoonday** [comment]
I am not quite fibre but I have a 35mbit download so it’s not exactly a slouch. 

ZHA is the integration for Zigbee of which I’m using a HUSBZB-1 single and a Hue hub with two light bulbs.

Approx 55 entities.

Zha - https://www.home-assistant.io/components/zha/

**Comment by vincenzosuraci** [comment]
There Is a HA integration to monitor the CPU load. I tried to run HA on a RPi1 and, during each startup, I experienced 100% CPU and Lovelace not responding for minutes and random hangs. So, yes, It could be that too short scan_intervals can affect the performances of RPi when dealing with number of devices.

**Comment by vincenzosuraci** [comment]
Since I guess we identified the source of the problem, we can declare closed this issue.

**Comment by tukadafoonday** [comment]
So getting back to this - I still have issues with the success of boot all the way into home assistant. The machine gets up and running as far as samba, ssh, etc; however, fails to get me into a webui.

After identifying it could have been the rpi being the issue I have since installed a fresh copy of HA on proxmox with 4 xeon cores and 5gb of ram on an older IBM server.

Still the same issue as the rpi.

logs are here - https://pastebin.com/Wb9fFLqk

I am very confused.

*****After some playing I have found a scan of about 150seconds to allow me to boot into the HA WebUI.  All of the entities for switches show they're off though, when infact they're actually on.******
